lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+            if promiscuous_mode_previous != self.sec_promiscuous_mode:
+                changed = True
+            if mac_changes_previous != self.sec_mac_changes:
+                changed = True
+            if forged_transmits_previous != self.sec_forged_transmits:
+                changed = True
+        return sec_spec, changed, promiscuous_mode_previous, mac_changes_previous, forged_transmits_previous
+
+    def check_trunk_vlan_config(self, pg_config):
+        """Check trunk VLAN config"""
+        changed_vlan_trunk_range = False
+        # NOTE: the following code is for backward compatibility only. Will be removed in 2.11
+        if self.vlan_trunk:
+            trunk_vlan_spec = vim.dvs.VmwareDistributedVirtualSwitch.TrunkVlanSpec()
+            vlan_id_start, vlan_id_end = self.vlan_id.split('-')
+            trunk_vlan_spec.vlanId = [vim.NumericRange(start=int(vlan_id_start.strip()), end=int(vlan_id_end.strip()))]
+            # Do not check if range is already configured; just classify it as changed
+            changed_vlan_trunk_range = True
+        else:
+            vlan_id_ranges = self.vlan_trunk_range
+            trunk_vlan_spec = vim.dvs.VmwareDistributedVirtualSwitch.TrunkVlanSpec()
+            vlan_id_list = []
+            for vlan_id_range in vlan_id_ranges:
+                vlan_id_range_found = False
+                vlan_id_start, vlan_id_end = self.get_vlan_ids_from_range(vlan_id_range)
+                if pg_config:
+                    # Check if range is already configured
+                    for current_vlan_id_range in pg_config.defaultPortConfig.vlan.vlanId:
+                        if (current_vlan_id_range.start == int(vlan_id_start) and
+                                current_vlan_id_range.end == int(vlan_id_end)):
+                            vlan_id_range_found = True
+                            break
+                    if vlan_id_range_found is False:
+                        changed_vlan_trunk_range = True
+                vlan_id_list.append(vim.NumericRange(start=int(vlan_id_start), end=int(vlan_id_end)))
+            if pg_config:
+                # Check if range needs to be removed
+                for current_vlan_id_range in pg_config.defaultPortConfig.vlan.vlanId:
+                    vlan_id_range_found = False
+                    for vlan_id_range in vlan_id_ranges:
+                        vlan_id_start, vlan_id_end = self.get_vlan_ids_from_range(vlan_id_range)
+                        if (current_vlan_id_range.start == int(vlan_id_start) and
+                                current_vlan_id_range.end == int(vlan_id_end)):
+                            vlan_id_range_found = True
+                            break
+                    if vlan_id_range_found is False:
+                        changed_vlan_trunk_range = True
+            trunk_vlan_spec.vlanId = vlan_id_list
+        return changed_vlan_trunk_range, trunk_vlan_spec
```

Comment:
Shouldn't this be in the reverse order, so it is in line with the other check methods? (consistency in code is good). That way you always use `result[0]` (will need to update that at other spots as well)

Reply:
Yes, the spec needs to be returned first to be similar with other modules. Thanks for pointing this out. I changed it.